frontend/src/cljs/imas_seamap/map/events.cljs

```diff
@@ -556,24 +556,49 @@
                  [:ui.catalogue/catalogue-add-nodes-to-layer category layer "cat" [:data_classification]]
                  [:map/pan-to-layer layer]])})
 
-(defn update-active-network [db [_ network]]
-  (if-not (= network (get-in db [:map :active-network]))
-    (-> db
-        (assoc-in [:map :active-park] nil)
-        (assoc-in [:map :active-network] network))
-    db))
-
-(defn update-active-park [{:keys [map] :as db} [_ {:keys [network] :as park}]]
-  (-> db
-      (assoc-in [:map :active-network] (first-where #(= (:name %) network) (:networks map)))
-      (assoc-in [:map :active-park] park)))
-
-(defn update-active-zone [db [_ zone]]
-  (-> db
-      (assoc-in [:map :active-zone] zone)
-      (assoc-in [:map :active-zone-iucn] nil)))
-
-(defn update-active-zone-iucn [db [_ zone-iucn]]
-  (-> db
-      (assoc-in [:map :active-zone-iucn] zone-iucn)
-      (assoc-in [:map :active-zone] nil)))
+(defn update-active-network [{:keys [db]} [_ network]]
+  (let [db (if-not (= network (get-in db [:map :active-network]))
+             (-> db
+                 (assoc-in [:map :active-park] nil)
+                 (assoc-in [:map :active-network] network))
+             db)]
```

Comment:
This actually _is_ a good use-case for `cond->` I think!  (that general pattern of "if X do this transform, else just return the thing".  It's not so great for "if X do this transform, otherwise do this other transform").  And although it can be a bit tricky to follow at first, I'm sure I've seen `->` used as that transform; eg

```clojure
(cond-> db
  (not= network (get-in db [:map :active-network]))
  (->
    (assoc-in [:map :active-park] nil)
    (assoc-in [:map :active-network] network)))
```

Reply:
Done, thanks!